packages/basti-cdk/src/basti-instance.ts

```diff
@@ -117,9 +117,33 @@ export class BastiInstance extends Construct implements IBastiInstance {
       subnetType: aws_ec2.SubnetType.PUBLIC,
     };
 
+    const bastiInstancePolicy = new aws_iam.PolicyDocument({
+      statements: [
+        new aws_iam.PolicyStatement({
+          actions: ['ec2:DescribeInstances'],
+          // ec2:DescribeInstances does not support resource-level permissions
+          resources: ['*'],
+        }),
+        new aws_iam.PolicyStatement({
+          actions: ['ec2:CreateTags'],
+          // This is broader than we would like, but it is required to tag
```

Comment:
This is unfortunately not possible as CDK is "compiled" to CloudFormation first, and then deployed. This means that regardless of the order in which they are defined, the role is created with all the IAM statements, even if those IAM statements are added after the creation of the instance.

In this case, what causes the circular dependency is the following:
1. The role is created (without policies)
2. The role is attached to the IAM instance (making the instance dependent on the creation of the role)
3. Then an IAM statement is added that contains the **token** to resolve the arn. Creating a dependency from the role to the instance)
4. When CDK tries to compile, and then deploy, it gives an error, as the creation of IAM policies depends on the creation of the instance. Which depends on the creation of the role.

Reply:
I think CDK must support this via the ref syntax. Let me check it quickly ⏲

Reply:
It worked! 

See [this Gist](https://gist.github.com/BohdanPetryshyn/e0484de355afdb21f30a09371693eb6f#file-basti-instance-ts-L162-L185)

Reply:
Here's the result of the `cdk synth` command: ([link](https://gist.github.com/BohdanPetryshyn/c1b7262c039c895220f3ac3cfcc41bdf#file-synth-yaml-L442-L465))

Here's the policy created:

<img width="1416" alt="image" src="https://github.com/BohdanPetryshyn/basti/assets/45905756/eb82af3c-5b6e-4e9f-b7cd-16f2d088d13f">

Reply:
Ah, you are creating a new policy and attaching it inline, and because the policy is its own construct it all works out. Yeah, that that would work. No clue why I didn't think of that.

Alright, I'll implement that